test/test_framework.py

```diff
@@ -17,39 +17,75 @@
 import gc
 import inspect
 import io
+import logging
 import os
+import pathlib
 import re
 import shutil
 import sys
 import tempfile
 import typing
 from pathlib import Path
-from test.test_helpers import BaseTestCase, fake_script
+from test.test_helpers import FakeScript
 from unittest.mock import patch
 
 import pytest
 
 import ops
 from ops.framework import _BREAKPOINT_WELCOME_MESSAGE, _event_regex
+from ops.model import _ModelBackend
 from ops.storage import JujuStorage, NoSnapshotError, SQLiteStorage
 
 
-class TestFramework(BaseTestCase):
-
-    def setUp(self):
-        self.tmpdir = Path(tempfile.mkdtemp())
-        self.addCleanup(shutil.rmtree, str(self.tmpdir))
-
-        patcher = patch('ops.storage.SQLiteStorage.DB_LOCK_TIMEOUT', datetime.timedelta(0))
-        patcher.start()
-        self.addCleanup(patcher.stop)
-
-    def test_deprecated_init(self):
+def create_model():
+    """Create a Model object."""
+    backend = _ModelBackend(unit_name='myapp/0')
+    meta = ops.CharmMeta()
+    model = ops.Model(meta, backend)
+    return model
+
+
+def create_framework(request: pytest.FixtureRequest,
+                     *,
+                     model: typing.Optional[ops.Model] = None,
+                     tmpdir: typing.Optional[pathlib.Path] = None):
+    """Create a Framework object.
+
+    By default operate in-memory; pass a temporary directory via the 'tmpdir'
+    parameter if you wish to instantiate several frameworks sharing the
+    same dir (e.g. for storing state).
+    """
+    if tmpdir is None:
+        data_fpath = ":memory:"
+        charm_dir = 'non-existant'
+    else:
+        data_fpath = tmpdir / "framework.data"
+        charm_dir = tmpdir
+
+    patcher = patch('ops.storage.SQLiteStorage.DB_LOCK_TIMEOUT', datetime.timedelta(0))
+    patcher.start()
+    framework = ops.Framework(
+        SQLiteStorage(data_fpath),
+        charm_dir,
+        meta=model._cache._meta if model else ops.CharmMeta(),
+        model=model)  # type: ignore
+    request.addfinalizer(framework.close)
+    request.addfinalizer(patcher.stop)
+    return framework
+
+
+@pytest.fixture
+def fake_script(request: pytest.FixtureRequest) -> FakeScript:
+    return FakeScript(request)
+
+
+class TestFramework:
+    def test_deprecated_init(self, caplog: pytest.LogCaptureFixture):
         # For 0.7, this still works, but it is deprecated.
-        with self.assertLogs(level="WARNING") as cm:
+        with caplog.at_level(logging.WARNING):
             framework = ops.Framework(':memory:', None, None, None)  # type: ignore
-        assert "WARNING:ops.framework:deprecated: Framework now takes a Storage not a path" in \
-            cm.output
+        assert "WARNING:ops.framework:deprecated: Framework now takes a Storage not a path" in [
+            f"{record.levelname}:{record.name}:{record.message}" for record in caplog.records]
```

Comment:
`caplog.records` only have the message, no level or module, so I made this change.

There is another way, which is to specify module and log level and only assert the message, not asserting the whole `RNING:ops.framework:deprecated:...`.

Same below.